src/x3f_extract.c

```diff
@@ -309,38 +310,57 @@ int main(int argc, char *argv[])
     }
 
     if (extract_jpg) {
-      if (X3F_OK != x3f_load_data(x3f, x3f_get_thumb_jpeg(x3f))) {
-	x3f_printf(ERR, "Could not load JPEG thumbnail from %s\n", infile);
+      if (X3F_OK != (ret = x3f_load_data(x3f, x3f_get_thumb_jpeg(x3f)))) {
```

Comment:
This is a quite common way to do it. And ... it was essentially the same in the old code. The only difference is that it extracts ret, to be used in the error printout.

I do not feel like doing some refactoring of the code for style's sake. We have bigger problems with this code than that.

This is a preparation for solving the sd Quattro firmware 1.04 bug and also preparing for supporting the the sd Quattro H. To get better error printouts so we see what happens.